controllers/grafanafolder_controller.go

```diff
@@ -400,6 +405,30 @@ func (r *GrafanaFolderReconciler) onFolderCreated(ctx context.Context, grafana *
 	return nil
 }
 
+func (r *GrafanaFolderReconciler) retrieveParentFolderUID(ctx context.Context, cr *grafanav1beta1.GrafanaFolder) (string, error) {
```

Comment:
Can you align this with the implementation in [the AlertRuleGroup Controller](https://github.com/grafana/grafana-operator/blob/master/controllers/grafanaalertrulegroup_controller.go#L335)

The main difference lies in the way it handles errors by saving them in the status

Reply:
Actually - can we move this into the shared controller logic and reuse it across this + the alert rule group controller? we'll need the same logic again in the dashboard soon so this could make sense

Reply:
(if you can't find a clean way, no worries - we have quite a few duplications across controllers for ergonomic reasons)

Reply:
I'll try to find a way to do this. Thank you for the improvment prposal.